Python_Scripts/plot_Tolerances_Main.py

```diff
@@ -1,278 +1,288 @@
 # Main Manuscript Plot --- Figure 4
 # script to create box plot with percentiles and median with failure rates
 
-import matplotlib.pyplot as plt
+import os
 import numpy as np
-from averageTime import *
-
-
-def Tolerances(Multistep_Method, prefix):
+import pandas as pd
+import matplotlib.pyplot as plt
 
-    # check whether the folder 'Benchmarking_of_numerical_ODE_integration_methods/Data' exists
-    if not os.path.exists('../../Benchmarking_of_numerical_ODE_integration_methods/Data/TolerancesStudy'):
-        tolerance_path = '../Data/TolerancesStudy/' + Multistep_Method
-    elif os.path.exists('../../Benchmarking_of_numerical_ODE_integration_methods/Data/TolerancesStudy'):
-        tolerance_path = '../../Benchmarking_of_numerical_ODE_integration_methods/Data/TolerancesStudy/' + Multistep_Method
+from averageTime import averaging
+from C import DIR_DATA_TOLERANCES
+
+Multistep_Method = 'BDF'
+prefix = '2_'
+
+tolerance_path = os.path.join(DIR_DATA_TOLERANCES, Multistep_Method)
+
+# main .tsv file to norm all other files
+main_tsv = pd.read_csv(os.path.join(
+    tolerance_path, prefix + '06_06.tsv'), sep='\t')
+
+# get new .tsv file
+main_tsv = averaging(main_tsv)
+
+# set two axes objects
+figure = plt.figure()
+ax1 = figure.add_axes([0.11, 0.55, 0.86, 0.4])  # ax = plt.axes()
+ax2 = figure.add_axes([0.11, 0.22, 0.86, 0.28])
+ax3 = figure.add_axes([0.11, 0.15, 0.86, 0.05])
+
+# get list for all data
+xTickLabel = []
+total_data = []
+
+# open all .tsv tolerance files
+tolerance_files_old = sorted(os.listdir(tolerance_path))
+# del tolerance_files[0]
+tolerance_files = []
+for iTolFile in range(0, len(tolerance_files_old)):
+    if len(tolerance_files_old[iTolFile].split(prefix)) > 2:
+        tolerance_files.append(
+            tolerance_files_old[iTolFile].split('_')[1] + '_' +
+            tolerance_files_old[iTolFile].split('_')[2])
+    else:
+        tolerance_files.append(tolerance_files_old[iTolFile].split(prefix)[1])
+
+###############################################################################
+# 1.PART: create Box Plot
+
+all_averaged_files = []
+tolerance_files.insert(6, '')
+tolerance_files.insert(13, '')
+tolerance_files.insert(20, '')
+tolerance_files.insert(27, '')
+tolerance_files.insert(34, '')
```

Comment:
Is this used for something like spacing, line breaks, or similar stuff?

Reply:
I think so, yes.